modules/stub/src/stub.js

```diff
@@ -1,218 +1,178 @@
-(function() {
+(function () {
+  const TCF_LOCATOR_NAME = '__tcfapiLocator';
 
-  const makeStub = () => {
-
-    const TCF_LOCATOR_NAME = '__tcfapiLocator';
+  function makeStub() {
     const queue = [];
-    let win = window;
+    const currentWindow = window;
+    let frameLocator = window;
+    let hasEventListener = false;
     let cmpFrame;
 
     function addFrame() {
+      const doc = currentWindow.document;
 
-      const doc = win.document;
-      const otherCMP = !!(win.frames[TCF_LOCATOR_NAME]);
-
-      if (!otherCMP) {
-
-        if (doc.body) {
-
+      if (!currentWindow.frames[TCF_LOCATOR_NAME]) {
+        if (doc.readyState === 'interactive' || doc.readyState === 'complete') {
```

Comment:
We don't want to trigger on readyState getting to `interactive`. Substantial testing was done on several alternatives and the `doc.body` exists and `setTimeout` was determined to be the earliest-possible method (despite it's hacky look) to get the iframe added, which is critical since it is part of CMP detection.

Reply:
@a2intl okay, should i revert the changes for the `document.body` detection or is this PR obsolete?

Reply:
my suggestion would be to close this PR and submit another with just the changes necessary to fix issue #166